In [1]:
import pandas as pd
import numpy as np
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)
df = pd.read_excel("ASHDEUR5.xlsx")
df.info()
print(df.columns.tolist())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4611 entries, 0 to 4610
Columns: 118 entries, IDCNTRY to SCOPE
dtypes: float64(108), int64(10)
memory usage: 4.2 MB
['IDCNTRY', 'IDPOP', 'IDGRADER', 'IDGRADE', 'WAVE', 'IDSCHOOL', 'IDCLASS', 'IDSTUD', 'ITLANG_HQ', 'LCID_HQ', 'ASBH01A', 'ASBH01B', 'ASBH01C', 'ASBH01D', 'ASBH01E', 'ASBH01F', 'ASBH01G', 'ASBH01H', 'ASBH01I', 'ASBH01J', 'ASBH01K', 'ASBH01L', 'ASBH01M', 'ASBH01N', 'ASBH01O', 'ASBH01P', 'ASBH01Q', 'ASBH01R', 'ASBH02A', 'ASBH02B', 'ASBH03A', 'ASBH03B', 'ASBH03C', 'ASBH03D', 'ASBH03E', 'ASBH03F', 'ASBH04', 'ASBH05AA', 'ASBH05AB', 'ASBH05B', 'ASBH06', 'ASBH07A', 'ASBH07B', 'ASBH07C', 'ASBH07D', 'ASBH07E', 'ASBH07F', 'ASBH07G', 'ASBH08A', 'ASBH08B', 'ASBH08C', 'ASBH08D', 'ASBH08E', 'ASBH08F', 'ASBH09', 'ASBH10', 'ASBH11A', 'ASBH11B', 'ASBH11C', 'ASBH11D', 'ASBH11E', 'ASBH11F', 'ASBH11G', 'ASBH11H', 'ASBH12', 'ASBH13', 'ASBH14A', 'ASBH14B', 'ASBH14C', 'ASBH15A', 'ASBH15B', 'ASBH16', 'ASBH17A', 'ASBH17B', 'ASBH18AA', 'ASBH18AB', 'A

## From a look at the codebook for this dataset, we can see that ASBH01 refers to, ASBH02 refers to  ....

Let's look at the ages these children came to the country.  **ASBH02B 1: Younger than 3 years old; 2: 3 to 5 years old; 3: 6 to 7 years old; 4: 8 years old or older**

I think it might actually be good to do an initial study of these new arrivals from all schools and then do a comparative analysis of a few schools after that. Let's see how many immigrant kids we have total (meaning those born in another country)

In [8]:
df["ASBH02B"].value_counts()

6.0    2317
2.0      75
9.0      60
1.0      53
3.0      36
4.0      32
Name: ASBH02B, dtype: int64

In [78]:
##Using shape[0] to count the rows.

In [77]:
number_immigrant_children = df[df["ASBH02B"]< 6.0].shape[0]
print(number_immigrant_children)

196


How many children spoke German before they started school?  **ASBH03A (1 is yes, 2 is no)**

In [10]:
df['ASBH03A'].value_counts()

1.0    2258
9.0     166
2.0     149
Name: ASBH03A, dtype: int64

2258 spoke German before they started school and 149 didn't. 

Let's see how many children spoke Arabic before they started school. **ASBH03B**
The other languages asked about in the test are Polish, Russian and Turkish, so we are primarily looking at Arabic speakers as potential refugees. 

In [11]:
df['ASBH03B'].value_counts()

2.0    2347
9.0     171
1.0      55
Name: ASBH03B, dtype: int64

Out of all the children asked 55 are native Arabic speakers. 

Let's find the Arabic speaking children who were not born in Germany.

In [31]:
arabic_speaking_pupils_IDs = df.query("ASBH03B==1.0")["IDSTUD"].to_list()

In [27]:
print(arabic_speaking_pupils_IDs)

[50010214, 50110317, 50180208, 50180211, 50180216, 50210316, 50250201, 50390103, 50400103, 50440104, 50510215, 50720310, 50750403, 50870204, 50870207, 51010213, 51060220, 51090111, 51110318, 51330208, 51370209, 51400202, 51420113, 51440306, 51440317, 51460211, 51460212, 51520103, 51600211, 51660303, 51720216, 51810312, 51860212, 51900106, 51910205, 51960204, 51960205, 51960207, 51960210, 51960213, 51960217, 51980107, 52030101, 52100212, 52250303, 52260303, 52310209, 52310217, 52310224, 52470104, 52490422, 52510311, 52510312, 52590107, 52660213]


In [79]:
arabic_speakers_df = df[df['IDSTUD'].isin(arabic_speaking_pupils_IDs)].copy()

In [80]:
arabic_speakers_df["ASBH02A"].value_counts()

2    37
1    17
9     1
Name: ASBH02A, dtype: int64

37 of our Arabic speakers were not born in Germany and 17 were. 

Let's look at the ages these children came to the country. ASBH02B 1: Younger than 3 years old; 2: 3 to 5 years old; 3: 6 to 7 years old; 4: 8 years old or older.

In [81]:
arabic_speakers_df["ASBH02B"].value_counts()

2    21
6    17
3     8
4     4
1     3
9     2
Name: ASBH02B, dtype: int64

## Let's look at how old these children are to check they are all the same age by reading in the School Context dataset which includes the column 'ASDAGE' so we can check this and also work out at what age each child arrived in Germany.

In [34]:
arabic_speaking_immigrant_IDs = arabic_speakers_df.query("ASBH02A==2.0")["IDSTUD"].to_list()

In [35]:
print(arabic_speaking_immigrant_IDs)

[50010214, 50110317, 50180208, 50180216, 50250201, 50390103, 50440104, 50510215, 50720310, 50750403, 51060220, 51110318, 51370209, 51400202, 51420113, 51440317, 51460211, 51460212, 51520103, 51600211, 51720216, 51810312, 51900106, 51910205, 51960204, 51960205, 51960207, 51960210, 51960213, 51960217, 51980107, 52100212, 52250303, 52260303, 52470104, 52510311, 52590107]


In [36]:
student_context_df = pd.read_excel("ASGDEUR5.xlsx")
student_context_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4611 entries, 0 to 4610
Columns: 155 entries, IDCNTRY to SCOPE
dtypes: float64(129), int64(26)
memory usage: 5.5 MB


In [55]:
arabic_speaking_immigrant_student_context_df = student_context_df[student_context_df['IDSTUD'].isin(arabic_speaking_immigrant_IDs)].copy()

In [63]:
##I have added .copy() because I don't want the 'setting with copy' warning. I have created a separate dataframe instead of a slice of the existing one. 

In [96]:
arabic_speaking_immigrant_student_context_df["ASDAGE"].astype(float).value_counts().sort_index(ascending=False)

12.33    1
12.00    1
11.67    1
11.58    1
11.42    1
11.25    3
11.17    1
11.08    1
11.00    1
10.75    1
10.67    1
10.42    3
10.33    5
10.25    3
10.17    4
10.00    2
9.92     1
9.83     2
9.75     1
9.67     3
Name: ASDAGE, dtype: int64

In [95]:
pd.options.display.float_format = '{:.2f}'.format

So it seems we have some 12 year olds so it seems they may have been put in lower grades or repeated the year.

I want to look at this group in a bit more detail in another notebook. So I will put them in a new excels and read them into a new notebook. 

In [104]:
arabic_speakers_df.to_excel("Arabic_speakers_HC.xlsx")

In [105]:
arabic_speaking_immigrant_student_context_df.to_excel("Arabic_speakers_SC.xlsx")